### PR TITLE
tools/printast: Print extended AST

### PR DESCRIPTION
**Pull request by Julow**
This is the equivalent of `ocamlc -dparsetree` for out "extended AST".

Usage:

    dune exec -- tools/printast/printast.exe test.ml

### CODE REVIEW

**Comment by gpetiot** [comment]
Good idea!

> This is the equivalent of `ocamlc -dparsetree` for out "extended AST".
> 
> Usage:
> 
> ```
> dune exec -- tools/printast/printast.exe test.ml
> ```

This should go in `tools/printast/README.md`

File: tools/printast/printast.ml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+let parse_and_print parsef printf lexbuf =
+  printf Format.std_formatter (parsef lexbuf)
+
+let get_ppf fname =
+  match Filename.extension fname with
+  | ".mli" ->
+      (parse_and_print Parse.interface Printast.interface, "interface")
+  | _ ->
+      ( parse_and_print Parse.implementation Printast.implementation
+      , "implementation" )
```
Comment (gpetiot) [inline tools/printast/printast.ml:18]: Maybe `Bin_conf.kind_of_ext` could be re-used (moved to `Syntax` maybe? since it returns a `Syntax.t`)

**Review by gpetiot — APPROVED** [review]
Looks good, thanks!
Maybe add a README to this tool as well, and good to go when the CI is green.

File: tools/printast/printast.ml (hunk 1 above)
Comment (Julow) [reply tools/printast/printast.ml]: Good idea! I also moved the match on the syntax to `Extended_ast`.

**Pull request merged**